dali/pipeline/operators/reader/loader/file_loader.cc

```diff
@@ -98,8 +98,8 @@ void FileLoader::ReadSample(ImageLabelWrapper* image_label) {
   auto image_pair = image_label_pairs_[current_index_++];
 
   // handle wrap-around
-  if (current_index_ == Size()) {
-    current_index_ = 0;
+  if (IsNextShard(current_index_)) {
+    Reset();
```

Comment:
is reset equivalent to current_index_ = 0 ?

Reply:
No, the whole point of that patch is to stick every GPU to its shard. So reset will assign current_index_ to appropriate value, usually != 0.